src/main/scala/part2foundations/SparkAPIs.scala

```diff
@@ -51,7 +52,7 @@ object SparkAPIs {
   rddTimes5.count() // ~20s
   // one stage
 
-  val dfTimes5 = df.select("id * 5 as id")
+  val dfTimes5 = df.select(expr("id * 5").as("id"))
   val dfTimes5Count = dfTimes5.selectExpr("count(*)")
```

Comment:
Replaced val dfTimes5 = df.select("id * 5 as id") with df.select(expr("id * 5").as("id")), because there is an issue with the column expression 'id * 5 as id' in code. Now it's running fine.

Reply:
Ran and tested all the Scala files. Everything is working fine.
Again I faced a few exceptions based on my OS's environmental issues,
CogroupingRDDs.scala, SimpleRDDJoins.scala, SparkJobAnatomy.scala, I2ITransformations.scala - Exception in thread "main" org.apache.hadoop.mapred.InvalidInputException: Input path does not exist.

SparkRecap.scala, JoinsRecap.scala,  - below exception is coming because, I'm trying to run a Spark program that uses Hadoop's native IO library on a Windows system. However, the NativeIO library is not compatible with Windows, and therefore getting this exception.
Exception in thread "main" java.lang.UnsatisfiedLinkError: 'boolean org.apache.hadoop.io.nativeio.NativeIO$Windows.access0(java.lang.String, int)'

Reply:
I'm not sure that's the explanation. Which JDK are you trying to run the applications with?

Reply:
I'm using JDK 15.

Reply:
So I've just downloaded JDK 11 and used it to run the above piece `val dfTimes5 = df.select("id * 5 as id")` anf got an exception like this:
```Exception in thread "main" org.apache.spark.sql.AnalysisException: [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `id * 5 as id` cannot be resolved. Did you mean one of the following? [`id`].;
'Project ['id * 5 as id]
+- Project [value#2 AS id#5]
   +- SerializeFromObject [input[0, int, false] AS value#2]
      +- ExternalRDD [obj#1]

	at org.apache.spark.sql.errors.QueryCompilationErrors$.unresolvedAttributeError(QueryCompilationErrors.scala:307)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.org$apache$spark$sql$catalyst$analysis$CheckAnalysis$$failUnresolvedAttribute(CheckAnalysis.scala:147)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis0$6(CheckAnalysis.scala:266)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis0$6$adapted(CheckAnalysis.scala:264)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:244)
```

Reply:
Whereas, when changing the code as this: `val dfTimes5 = df.select(expr("id * 5").as("id"))` running the scala file successfully. therfore I had changed code from `val dfTimes5 = df.select("id * 5 as id")`  to  `val dfTimes5 = df.select(expr("id * 5").as("id"))`

Reply:
Is this fine or shall I roll back to the previous code ?

Reply:
This is too much hassle, you can just keep the old code